src/RA_MemManager.cpp

```diff
@@ -103,23 +103,27 @@ unsigned char MemManager::ActiveBankRAMByteRead(ra::ByteAddress nOffs) const
 {
     const BankData* bank = nullptr;
 
-    int bankID = 0;
-    const int numBanks = m_Banks.size();
+    auto bankID = 0U;
+    const auto numBanks = m_Banks.size();
     while (bankID < numBanks)
     {
         bank = &m_Banks.at(bankID);
+        Expects(bank != nullptr);
         if (nOffs < bank->BankSize)
+        {
+            Ensures(bank != nullptr);
```

Comment:
Clearly I need a tutorial on Expects and Ensures. Expects clearly behaves like an assert, but I'm assuming it also throws an execption on a release build. What does Ensures do? We just validated it wasn't null two lines ago, why would it be null here?

Reply:
It's for pre/post-conditions. If it fails the program terminates. They are basically the same. The rule says you have to test it for nullness on all paths. I'll see if wrapping the whole thing in an if statement does anything to suppress it.